__test__/actions/authActions.test.js

```diff
@@ -0,0 +1,143 @@
+
+  it('should create the REGISTER_WITH_EMAIL_FAILURE action if the api request was not successful', async () => {
+    mock.onPost('/register').reply(400, { message: 'Invalid registeration data' });
+
+    const expectedActions = [
+      { type: REGISTER_WITH_EMAIL_REQUEST },
+      { type: REGISTER_WITH_EMAIL_FAILURE, payload: 'Invalid registeration data' }
+    ];
+
+    await store.dispatch(registerWithEmail(badUserRegData, { closeModal, history }));
+    expect(store.getActions()).toEqual(expectedActions);
+  });
+
+  it('should create NETWORK_FAILURE action if there was a network error', async () => {
+    mock.onPost('/register').timeout(0);
+
+    const expectedActions = [
+      { type: REGISTER_WITH_EMAIL_REQUEST },
+      { type: NETWORK_FAILURE, payload: 'Connection error.' }
+    ];
+    await store.dispatch(registerWithEmail(badUserRegData, { closeModal, history }));
+    expect(store.getActions()).toEqual(expectedActions);
+  });
+
+  it('should create the REGISTER_WITH_EMAIL_SUCCESS action if the api request was successful', async () => {
+    mock.onPost('/register').reply(201, expectedResponseDataReg);
+
+    const expectedActions = [
+      { type: REGISTER_WITH_EMAIL_REQUEST },
+      { type: REGISTER_WITH_EMAIL_SUCCESS },
+      { type: SET_CURRENT_USER, user: false }
+    ];
+
+    await store.dispatch(registerWithEmail(rightUserRegData, { closeModal, history }));
+    expect(store.getActions()).toEqual(expectedActions);
+  });
+
+  it('should create VERIFY_USER_FAILURE action if the api request was not successful', async () => {
+    mock.onPatch(`/users/verify?query=${token}`).reply(401, {
+      message: 'Access denied, you are not authorized to access this route'
+    });
+
+    const expectedActions = [
+      { type: VERIFY_USER_REQUEST },
+      {
+        type: VERIFY_USER_FAILURE,
+        payload: 'Access denied, you are not authorized to access this route'
+      }
+    ];
+
+    await store.dispatch(verifyUser(token, { history }));
+    expect(store.getActions()).toEqual(expectedActions);
+  });
+
+  it('should create VERIFY_USER_FAILURE action if there was a network error', async () => {
+    mock.onPatch(`/users/verify?query=${token}`).timeout(0);
+
+    const expectedActions = [
+      { type: VERIFY_USER_REQUEST },
+      { type: NETWORK_FAILURE, payload: 'Connection error.' }
+    ];
+    await store.dispatch(verifyUser(token, { history }));
+    expect(store.getActions()).toEqual(expectedActions);
+  });
+
+  it('should create VERIFY_USER_SUCCESS action if the api request was successful', async () => {
+    mock.onPatch(`/users/verify?query=${token}`).reply(200, expectedResponseDataReg);
+
+    const expectedActions = [
+      { type: VERIFY_USER_REQUEST },
+      { type: VERIFY_USER_SUCCESS }
+    ];
+
+    await store.dispatch(verifyUser(token, { history }));
+    expect(store.getActions()).toEqual(expectedActions);
+  });
+
+  it('shoild create the SIGN_OUT_USER action when the user signs out', () => {
```

Comment:
I believe this should be 'SHOULD' not 'shoild'

Reply:
Thanks. I've corrected it